telegram/_bot.py

```diff
@@ -1998,9 +1999,22 @@ async def send_media_group(
         connect_timeout: ODVInput[float] = DEFAULT_NONE,
         pool_timeout: ODVInput[float] = DEFAULT_NONE,
         api_kwargs: JSONDict = None,
+        group_caption: Optional[str] = None,
+        # Notes on group_caption_parse_mode:
+        # 1. To be fully consistent with media items it would be called
+        # `group_parse_mode`, but that is ambiguous.
+        # 2. Typing ODVInput[str] = DEFAULT_NONE will make tests for shortcuts fail
+        # (check_defaults_handling) despite it being used for parse mode for individual captions.
```

Comment:
Regarding naming, I'd also be okay with dropping the `group_` prefix. What ambiguity do you see with `group_parse_mode`?

Regarding defaults: I hadn't thought of that yet, tbh, but we should make sure that defaults are properly applied. IISC line 2008 doesn't need to be changed for that, but we should in any case add tests cases that make sure of it. If you need help with that, please feel free to reach out (unit tests can be tricky in general and for the defaults-stuff in particular :D) Maybe [this unit test](https://github.com/python-telegram-bot/python-telegram-bot/blob/eb818da93f3df2b439f5aef6065add232a5106e8/tests/test_inputmedia.py#L574-L584) can give you an idea how we currently tests the defaults setpu in the unit tests.

Reply:
Thank you for the feedback and the encouraging comment! :blush:

I'll start working on this and other comments, but a quick note regarding `parse_mode`. I thought that in a context of a single media item `parse_mode` was sort of more obvious than in a context of a group. Maybe `parse_mode` should also be `caption_parse_mode`, but at least there are not so many arguments for a single media item than there are for a group. Adding `_caption_` makes it clearer that these 3 arguments sort of belong together.

BTW I see that I put `group_parse_mode` (not `group_caption_parse_mode`) into the Note in docstring, noted.

Anyway, if you think that `group_parse_mode` won't be confusing for users, I'll be happy to make the argument name consistent with that of a single media item.

Reply:
Mh, my preferred rule of naming here would just be that the parameters that this PR introduces to `send_media_group` should not differ from the parameters of `InputMedia*`. that would give us `parse_mode`, `caption` and `caption_entities`, possibly with a `group_` prefix (although my I personally don't see a big advantage of having that prefix). IMO the correspondence to the established parameters of `InputMedia*` should be enough to avoid confusion …

Reply:
OK, I see your logic. I also forget that the project has great docs 😆 

Renaming all arguments to match the arguments of functions that handle individual media (commit 61af4f491fe5390b3f9e30ac28553366531af759)

At least I think it was not such a bad idea to name arguments differently in the beginning because renaming in this direction is much easier than in the opposite one.

> We should make sure that defaults are properly applied. <...> we should in any case add tests cases that make sure of it. If you need help with that, please feel free to reach out. Maybe [this unit test](https://github.com/python-telegram-bot/python-telegram-bot/blob/eb818da93f3df2b439f5aef6065add232a5106e8/tests/test_inputmedia.py#L574-L584) can give you an idea how we currently tests the defaults setup in the unit tests.

I must admit I'm having trouble understanding this, but I guess from pedagogical point of view it would be best for me to keep trying for another day before crying for help 😆

Reply:
> Take your time with the defaults stuff, but feel free to reach out for help if you want it 

I guess I understand better **how** these tests work; what I don't understand is **why** we need a test for something that defaults to `None` rather than `DEFAULT_NONE`. 

The only two existing unit tests check arguments that default to `DEFAULT_NONE`. So if we decide to leave `parse_mode: Optional[str] = None` (as opposed to `parse_mode: ODVInput[str] = DEFAULT_NONE`) then I guess I can't understand why an additional check is needed for `parse_mode` (and maybe you meant also for `caption` and `caption_entities`).

Reply:
I think I see what you mean. 
Still no regrets you let me work on this issue? 🤣

Reply:
> Still no regrets you let me work on this issue? 🤣

Not at all 🙆‍♂️

Reply:
Added commit b0d20c4dfda80095f644c1ba6542db7fc4674f07
I decided to put `DEFAULT_NONE` for `parse_mode` although I remember you said [here](https://github.com/python-telegram-bot/python-telegram-bot/pull/3295#discussion_r997365686) it was not necessary. But it seems the right way.

What possibly worries me:
- to make tests of shortcuts pass, I had to exclude `parse_mode` from [one check](https://github.com/python-telegram-bot/python-telegram-bot/commit/b0d20c4dfda80095f644c1ba6542db7fc4674f07#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128) in `conftest.py`. It kind of makes sense because `parse_mode` for the group is not a "regular argument" but I still worry that I did something wrong just to avoid an exception being thrown
- I tried to combine tests of defaults (right now I added two separate test functions `test_send_media_group_default_parse_mode` and `test_send_media_group_default_parse_mode_overridden`) in one test function along the lines of `test_send_media_group_default_protect_content`. If do one combined test function it makes the first call to `send_media_group()` just fine but then throws `ValueError` at the second call, as if one of the items had a caption (when in fact, and I have checked, the fixture group was not changed between calls). I suspect I'm missing something tricky here so I'll continue to look into that.